Src/DependencyCollector/Shared/ServiceBusDiagnosticsEventHandler.cs

```diff
@@ -0,0 +1,171 @@
+    {
+        public const string DiagnosticSourceName = "Microsoft.Azure.ServiceBus";
+        private const string StatusPropertyName = "Status";
+        private const string EntityPropertyName = "Entity";
+        private const string EndpointPropertyName = "Endpoint";
+
+        private readonly TelemetryClient telemetryClient;
+
+        // Every fetcher is unique for event payload and particular property into this payload
+        // when we first receive an event and require particular property, we cache fetcher for it
+        // There are just a few (~10) events we can receive and each will a have a few payload fetchers
+        // I.e. this dictionary is quite small, does not grow up after service warm-up and does not require clean up
+        private readonly ConcurrentDictionary<Property, PropertyFetcher> propertyFetchers = new ConcurrentDictionary<Property, PropertyFetcher>();
+
+        internal ServiceBusDiagnosticsEventHandler(TelemetryConfiguration configuration)
+        {
+            this.telemetryClient = new TelemetryClient(configuration);
+        }
+
+        public bool IsEventEnabled(string evnt, object arg1, object arg2)
+        {
+            return !evnt.EndsWith(TelemetryDiagnosticSourceListener.ActivityStartNameSuffix, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public void OnEvent(KeyValuePair<string, object> evnt, DiagnosticListener ignored)
+        {
+            Activity currentActivity = Activity.Current;
+
+            switch (evnt.Key)
+            {
+                case "Microsoft.Azure.ServiceBus.ProcessSession.Stop":
+                case "Microsoft.Azure.ServiceBus.Process.Stop":
+                    this.OnRequest(evnt.Key, evnt.Value, currentActivity);
+                    break;
+                case "Microsoft.Azure.ServiceBus.Exception":
+                    break;
+                default:
+                    this.OnDependency(evnt.Key, evnt.Value, currentActivity);
+                    break;
+            }
+        }
+
+        private void OnDependency(string name, object payload, Activity activity)
+        {
+            DependencyTelemetry telemetry = new DependencyTelemetry { Type = RemoteDependencyConstants.AzureServiceBus };
+
+            this.SetCommonProperties(name, payload, activity, telemetry);
+            
+            // Endpoint is URL of particular ServiceBus, e.g. sb://myservicebus.servicebus.windows.net/
+            string endpoint = this.FetchPayloadProperty<Uri>(name, EndpointPropertyName, payload)?.ToString();
+
+            // Queue/Topic name, e.g. myqueue/mytopic
+            string queueName = this.FetchPayloadProperty<string>(name, EntityPropertyName, payload);
+
+            // Target uniquely identifies the resource, we use both: queueName and endpoint 
+            // with schema used for SQL-dependencies
+            telemetry.Target = string.Join(" | ", queueName, endpoint);
```

Comment:
should it be endpoint followed by queuename to be consistent with sql. (sqlserverinstancename | dbname)

Reply:
agree